cmd/kubeops/internal/handler/handler.go

```diff
@@ -91,31 +94,21 @@ func WithHandlerConfig(storageForDriver agent.StorageForDriver, options Options)
 				return
 			}
 
-			// System configuration and clients, using the service serviceaccount
-			// Used to retrieve apprepositories and secrets related to them
-			svcRestConfig, err := rest.InClusterConfig()
-			if err != nil {
-				log.Errorf("Failed to create in-cluster config with service account: %v", err)
-				response.NewErrorResponse(http.StatusInternalServerError, authUserError).Write(w)
-				return
-			}
-			svcKubeClient, err := kubernetes.NewForConfig(svcRestConfig)
-			if err != nil {
-				log.Errorf("Failed to create kube client with service account: %v", err)
-				response.NewErrorResponse(http.StatusInternalServerError, authUserError).Write(w)
-				return
+			kubeappsNamespace := os.Getenv("POD_NAMESPACE")
+			if namespace == "" {
+				namespace = defaultNamespace
```

Comment:
Why are we defaulting/guessing that the Kubeapps namespace is the `kube-system` here (`metav1.NamespaceSystem`)?

We should now be able to fail to start, given that `POD_NAMESPACE` is set for both tiller-proxy and kubeops? EDIT: Ah, because this is within a function which has no option of returning an error :/ . If `WithHandlerConfig` requires knowing what the kubeapps namespace is, let's make it a required arg?
EDIT2: And because you've not changed this, it was already like this in `pkg/chart` from where it was moved. It seems like a bad guess to make, potentially a security issue.

My vote would be to pass it as a required arg (and in the main.go, fail if it's not defined). Let me know if you've got other ideas.